programs/drift/src/controller/spot_position.rs

```diff
@@ -93,6 +93,16 @@ pub fn update_spot_balances_and_cumulative_deposits(
         }
     }
 
+    if spot_position.balance_type == SpotBalanceType::Borrow
```

Comment:
if an asset was already being short sold and the asset tier switched to `Protected`, this would prevent a user from partially closing their borrow?

Reply:
yeah, just gotta check update direction